packages/autocomplete-js/src/types/index.ts

```diff
@@ -1,14 +1,14 @@
+export * from '@algolia/autocomplete-shared/dist/esm/js';
 export * from './AutocompleteApi';
-export * from './AutocompleteClassNames';
-export * from './AutocompleteCollection';
-export * from './AutocompleteComponents';
 export * from './AutocompleteDom';
-export * from './AutocompleteOptions';
-export * from './AutocompletePlugin';
-export * from './AutocompletePropGetters';
-export * from './AutocompleteRender';
-export * from './AutocompleteRenderer';
-export * from './AutocompleteSource';
-export * from './AutocompleteState';
-export * from './AutocompleteTranslations';
-export * from './HighlightHitParams';
+
+import {
+  AutocompleteOptions as AutocompleteCoreOptions,
+  BaseItem,
+} from '@algolia/autocomplete-core';
+import { AutocompleteOptions as AutocompleteJsOptions } from '@algolia/autocomplete-shared/dist/esm/js';
+
+export interface AutocompleteOptions<TItem extends BaseItem>
+  extends AutocompleteJsOptions<TItem> {
+  insights?: AutocompleteCoreOptions<TItem>['insights'];
+}
```

Comment:
We augment the `AutocompleteOptions` type with the `insights` type option here because augmenting it in the `autocomplete-shared` package would create a circular dependency.